Assets/SEE/Layout/NodeLayouts/IncrementalTreeMapLayout.cs

```diff
@@ -0,0 +1,354 @@
+namespace SEE.Layout.NodeLayouts
+{
+    /// <summary>
+    /// A node layout designed for the evolution setting.
+    /// Incremental Tree Map Layout is a Tree Map Layout,
+    /// that can guarantee stability in the layouts over the series of graphs in the evolution
+    /// while not neglecting the aspect of visual quality either.
+    /// </summary>
+    public class IncrementalTreeMapLayout : HierarchicalNodeLayout, IIncrementalNodeLayout
+    {
+        /// <summary>
+        /// Constructor.
+        /// </summary>
+        /// <param name="groundLevel">the y co-ordinate setting the ground level</param>
+        /// <param name="width">width of the rectangle in which to place all nodes in Unity units</param>
+        /// <param name="depth">width of the rectangle in which to place all nodes in Unity units</param>
+        /// <param name="settings">the settings for the layout</param>
+        public IncrementalTreeMapLayout(float groundLevel,
+            float width,
+            float depth,
+            IncrementalTreeMapSetting settings)
+            : base(groundLevel)
+        {
+            name = "IncrementalTreeMap";
+            _width = width;
+            _depth = depth;
+            _settings = settings;
+        }
+
+        /// <summary>
+        /// The adjustable parameters for the layout.
+        /// </summary>
+        private readonly IncrementalTreeMapSetting _settings;
+
+        /// <summary>
+        /// The width of the rectangle in which to place all nodes in Unity units.
+        /// </summary>
+        private readonly float _width;
+
+        /// <summary>
+        /// The depth of the rectangle in which to place all nodes in Unity units.
+        /// </summary>
+        private readonly float _depth;
+
+        /// <summary>
+        /// The node layout we compute as a result.
+        /// </summary>
+        private readonly Dictionary<ILayoutNode, NodeTransform> _layoutResult = new();
+
+        /// <summary>
+        /// A map to find a node (fast) by its ID
+        /// </summary>
+        private readonly Dictionary<string, Node> _nodeMap = new();
+
+        /// <summary>
+        /// A map to find a ILayoutNode (fast) by its ID
+        /// </summary>
+        private readonly Dictionary<string, ILayoutNode> _iLayoutNodeMap = new();
+
+        /// <summary>
+        /// The layout of the last revision in the evolution. Can be null.
+        /// </summary>
+        private IncrementalTreeMapLayout _oldLayout;
+
+        public IIncrementalNodeLayout OldLayout
+        {
+            set
+            {
+                if (value is IncrementalTreeMapLayout layout)
+                {
+                    this._oldLayout = layout;
+                }
+                else
+                {
+                    this._oldLayout = null;
+                    Debug.LogWarning("Incremental layout of last revision is not from same type " +
+                                     "as the layout for this revision");
```

Comment:
This seems like it should cause an exception instead, as it indicates a programming mistake rather than a user error, right?

Reply:
Yes, fixed in c6e4362fac4936ed214b6671d67cf730bddb70e6